provisioning/miniprov/hare_mp/cdf.py

```diff
@@ -37,11 +38,52 @@ def _create_node_descriptions(self) -> List[NodeDesc]:
             nodes.append(self._create_node(node))
         return nodes
 
+    def _create_pool_descriptions(self) -> List[PoolDesc]:
+        pools: List[PoolDesc] = []
+        conf = self.provider
+        cluster_id = conf.get('cluster>cluster_id')
+        storage_set_count = int(
+            conf.get(f'cluster>{cluster_id}>site>storage_set_count'))
+
+        for x in range(storage_set_count):
+            storage_set_name = conf.get(
+                f'cluster>{cluster_id}>storage_set{x+1}>name')
+            pools.append(PoolDesc(
+                name=Text(storage_set_name),
+                disk_refs=Maybe(DList([
+                    DiskRef(path=Text(device), node=Maybe(Text(node), 'Text'))
+                    for node in conf.get(
+                        f'cluster>{cluster_id}>storage_set{x+1}>server_nodes')
```

Comment:
We are not catching any exception here, what if there does not exist a `cluster>{cluster_id}>storage_set{x+1}>` key?

Reply:
Going to add new class for exception and handle all key related exception using that class.
Doing that in other task.

Reply:
Okay please create a ticket for the same and update here.

Reply:
Okay, kindly create a ticket and update here please.

Reply:
https://jts.seagate.com/browse/EOS-18813